subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformExecutionBuildOperationIntegrationTest.groovy

```diff
@@ -176,6 +205,371 @@ class ArtifactTransformExecutionBuildOperationIntegrationTest extends AbstractIn
+                @PathSensitive(PathSensitivity.RELATIVE)
+                @IgnoreEmptyDirectories
+                @NormalizeLineEndings
+                @InputArtifact
+                abstract Provider<FileSystemLocation> getInputArtifact()
+
+                @Classpath
+                @InputArtifactDependencies
+                abstract FileCollection getInputArtifactDependencies()
+
+                void transform(TransformOutputs outputs) {
+                    def input = inputArtifact.get().asFile
+                    println "processing [\${input.name}]"
+                    assert input.file
+                    def output = outputs.file(input.name + ".green")
+                    if (input.file) {
+                        output.text = input.text + ".green"
+                    } else {
+                        output.text = "missing.green"
+                    }
+                }
+            }
+        """
+
+        buildFile << """
+            project(":consumer") {
+                dependencies {
+                    implementation project(":producer")
+
+                    artifactTypes {
+                        jar {
+                            attributes.attribute(color, 'blue')
+                        }
+                    }
+                }
+            }
+        """
+        def producerTransformSpec = { Map<String, Object> it -> it.artifactName == 'producer.jar' }
+
+        when:
+        withBuildCache().run ":consumer:resolve"
+        then:
+        def projectExecution = findTransformExecution(producerTransformSpec)
+        with(snapshotInputsOperation(projectExecution).result) {
+            hash != null
+            classLoaderHash != null
+            implementationClassName == 'MakeGreen'
+            inputValueHashes.keySet() ==~ ['inputArtifactPath', 'inputPropertiesHash']
+            outputPropertyNames == ['outputDirectory', 'resultsFile']
+            inputFileProperties.keySet() ==~ ['inputArtifact', 'inputArtifactDependencies']
+            with(inputFileProperties.inputArtifactDependencies) {
+                hash != null
+                attributes == ['DIRECTORY_SENSITIVITY_DEFAULT', 'FINGERPRINTING_STRATEGY_CLASSPATH', 'LINE_ENDING_SENSITIVITY_DEFAULT']
+                with(Iterables.getOnlyElement(roots)) {
+                    hash != null
+                    path.endsWith('test-4.2.jar')
+                }
+            }
+            with(inputFileProperties.inputArtifact) {
+                hash != null
+                attributes == ['DIRECTORY_SENSITIVITY_IGNORE_DIRECTORIES', 'FINGERPRINTING_STRATEGY_RELATIVE_PATH', 'LINE_ENDING_SENSITIVITY_NORMALIZE_LINE_ENDINGS']
+                with(Iterables.getOnlyElement(roots)) {
+                    hash != null
+                    path.endsWith('producer.jar')
+                }
+            }
+        }
+
+        when:
+        run 'clean'
+        withBuildCache().run ':consumer:resolve'
+        then:
```

Comment:
💭  I think it would be interesting to add an assertion about the task outcome as well. Here, and for all other places where a task is executed with the build cache.

Reply:
The resolve task will always be executed, since it doesn't have outputs. What is the interesting part about the task outcome here?

Reply:
Ah, you are right. I thought it may be FROM-CACHE as well.